### ISSUE TITLE
select2 enabled field

### ISSUE THREAD

**Issue by achchu93** [elegantthemes/create-divi-extension#211]
### Problem Description
I wanna keep a select field with select2 js enabled in my custom module settings. then user can easily type and search the value within the dropdown. but currently it is not possible to do. Anyone can guide me please?

**Comment by lots0logs** [comment]
I'm afraid custom fields for module settings are not yet supported. We are planning to add support soon, hopefully sometime next month.

**Comment by xxtesaxx** [comment]
If you are fine with single select values, maybe you could write some JavaScript to listen for DOM changes, check if your modules settings are shown, find the select and apply select2 to it.

**Comment by achchu93** [comment]
> If you are fine with single select values, maybe you could write some JavaScript to listen for DOM changes, check if your modules settings are shown, find the select and apply select2 to it.

Yes, i did in that way. But what happens is, i initiated the select2 and then select the values and saved it. Then when i open it again the values are not set. Values are resetting from the actual select.

So if you can help me where does this values store? When we save, the ajax call has all the data. So i checked some places and edited the values. But no purpose of it.  These are the places i found values stores,
1. ETBuilderBackendDynamic.shortcode_object
2. ETBuilderBackend.shortcodeObject
3. ETBuilderBackend.shortcode_object

But even if i change the values in these objects that cant help. So if i know where the result store, then i can modify them.

I hope you understand the issue. This is not happening for backend editor. Only for ReactJS enabled editors.

**Comment by xxtesaxx** [comment]
If you send me a copy of your module I will check it out and see if I can find a solution

**Comment by alpezed** [comment]
Is it planned yet?

**Comment by CRessl** [comment]
Any Update on this? I noticed that the behaviour of the deleting value only happens in the "New and improved Version" not in the "classic" Version of the Editor. I also don't understand how the values are saved. If I look at the code, the values from a select field with select2 or selectize are set correctly, but after 1 sec it gets resetted to default " ".  In the bundle.js I couldn't find an interval function or something similar which could cause this. Only a listener for element changes...

Furthermore the current select fields perfomance with over 500 items is absolutely bad. In my case I need to dynamically get all posts from the past two years with a search function. The loading time after I clicked on the select field is massive. The previous handling with these fields was way better in my opinion
